### ISSUE TITLE
specify custom iteration count "p2c" for PBES2, seems fixed to 8192

### ISSUE THREAD

**Issue by gaiuscosades** [latchset/jwcrypto#279]
When for instance generating a password bound JWE object with {"alg": "PBES2-HS512+A256KW"} i do not find a way to specify a custom iteration count, 8192 is always used. Same i guess would be true for "p2s" although I see less common reasons one might would like to supply this parameter with a special value.

I tried to specify the parameter when generating the object using a header dictionary with the desired value in it e.g. {"p2c": 100000} but this crashes the header merge with

jwcrypto.jwe.InvalidJWEData: Duplicate header: "p2c"

Is this behavior somehow well defined or is it just not yet implemented, to be able to deal with user given values?

Hower thanks for this great library!

**Comment by simo5** [comment]
Yeah it is currently hardcoded here:
https://github.com/latchset/jwcrypto/blob/7a181049f1b0b5ca846e97156310d181c4cce8da/jwcrypto/jwa.py#L612

I will mark this as a bug, given RFC 7518 does say that P2s and P2c are input parameters: https://datatracker.ietf.org/doc/html/rfc7518#section-4.8.1

**Comment by simo5** [comment]
@gaiuscosades can you check #280 ?

**Comment by gaiuscosades** [comment]
Thank you!

Seems to work fine for me now.

while testing i found a missing whitespace after "parameter" in jwe.py:283
but I do not want to open a separate issue for that, but wont leave it unmentioned. ^^

**Comment by simo5** [comment]
Thanks, I'll close this issue once the bugfix is merged.
